Hazelnut/premake5.lua

```diff
@@ -27,6 +27,11 @@ project "Hazelnut"
 	{
 		"Hazel"
 	}
+	
+	postbuildcommands
+	{
+		"{COPY} %{LibraryDir.VulkanSDK_DLL} %{wks.location}/bin/" .. outputdir .. "/%{prj.name}"
+	}
```

Comment:
Doesn't this only need to be done for DEBUG build?
For the destination,  does replacing  `%{wks.location}/bin/" .. outputdir .. "/%{prj.name}`    with   `%{cfg.targetdir}` work?  The latter is more succinct.

Reply:
Also, should Sandbox be done as well as Hazelnut?

Reply:
Right now, I haven't accounted what happens with the RELEASE build. So someone might wanna take a look at that.
I will test out `%{cfg.targetdir}`, and see what's happening.

Yes, this should also be there for Sandbox. But Sandbox hasn't been touched around since ages. So I am skeptical to go and make changes there for the time being.

Reply:
`%{cfg.targetdir}` works on my end, for 'Debug' build mode

Reply:
Nice.  I suggest to move the whole `postbuildcommands` bit so that it is part of the 	`filter "configurations:Debug"` bit.
Then test that it still works (for both Debug and Release builds)

Also, I think the same changes should be made for the Sandbox project (even though sandbox has other issues at present.. nevertheless it would be good to tidy up all this postbuildstuff in the one PR)

Reply:
So, moved the `postbuildcommands` to `filter "configurations:Debug"` and it works. Both Debug and Release builds are running properly. I checked the output directories, and Debug has those DLLs copied, whereas the Release doesn't have any DLL copied. So I believe Release builds will work fine without those libraries.

Reply:
@freeman40  You can check out the latest commit, and if it's up-to the expectations, please mark this message thread as resolved.

Reply:
I think it wouldn't hurt to put `"` around the targetdir to account for whitespaces in the location, similar to #449?

Reply:
I'm not sure you need " in this case.  449 is creating the post build step command line directly, whereas in this PR the post build command line is generated by the premake {COPY} token, which should be smart enough to deal with white spaces in the paths.  (I will test this to make sure)

Reply:
So is `"` needed? Should I add it to the targetdir?

Reply:
I just experimented with it, it would work for me only if both the source and destination arguments of the copy are enclosed in ""

Reply:
So it should be `"{COPY} "%{LibraryDir.VulkanSDK_DebugDLL}" "%{cfg.targetdir}""` ?

Reply:
Yes, Pikachuxxxx is correct, the " are needed.  I also just learnt that premake {COPY} is deprecated in favour of {COPYDIR}  (or {COPYFILE} if you are copying files.
So, it should be:
"{COPYDIR} \"%{LibraryDir.VulkanSDK_DebugDLL}\" \"%{cfg.targetdir}\""

Reply:
I just tried COPYDIR and it does not work, premake does not covert it, but COPY works fine
"{COPY} \"%{LibraryDir.VulkanSDK_DebugDLL}\" \"%{cfg.targetdir}\""
this is what i did.

Reply:
yes, COPYDIR is only available from in premake5.0.0-alpha16

Reply:
So the new python script to get premake downloads the latest version, but only if you dont have it already, there is no version check. 
but since COPY does the trick we should go with that one.

Reply:
I have merged the commits from upstream master to this branch. This puts this PR on the same page with the commit where PR #441 is merged. In PR #441, the python script downloads the premake5.exe binary, and the version is set to alpha16. So hopefully, the COPYDIR and COPY issue gets resolved.

For those who have cloned the repository earlier, from time to time as new features/setup functionalities get added, they need to execute those batch scripts to download the necessary updated stuff manually, on their own local machine. We cannot control every aspect of setup - some of it needs to be handled by the end user itself. We can add utilities to automate some aspects of upgrading, but the execution of it can, and should be, done by the user itself.

Another alternate feature would be to check which version of premake is being used in the premake scripts, and then add conditional statements, but that would clutter up stuff unnecessarily.

Reply:
I also tested this, copydir works perfect, the new exe gets downloaded, this is no issue anymore, ready to be merged 👍